framework/freedomotic-core/src/test/java/com/freedomotic/environment/GraphEdgeTest.java

```diff
@@ -0,0 +1,219 @@
+
+  @Test
+  public void equalsInputNotNullOutputFalse() {
+
+    // Arrange
+    final GraphEdge objectUnderTest = new GraphEdge(-301_989_904, 134_217_728, null);
+    final GraphEdge other = new GraphEdge(-301_989_904, null, -2_147_483_647);
+
+    // Act
+    final boolean retval = objectUnderTest.equals(other);
+
+    // Assert result
+    Assert.assertEquals(false, retval);
+  }
+
+  @Test
+  public void equalsInputNotNullOutputFalse2() {
+
+    // Arrange
+    final GraphEdge objectUnderTest = new GraphEdge(-301_989_890, null, null);
+    final GraphEdge other = new GraphEdge(null, -2_147_483_647, -2_147_483_647);
+
+    // Act
+    final boolean retval = objectUnderTest.equals(other);
+
+    // Assert result
+    Assert.assertEquals(false, retval);
+  }
+
+  @Test
+  public void equalsInputNotNullOutputFalse3() {
+
+    // Arrange
+    final GraphEdge objectUnderTest = new GraphEdge(-303_054_864, 134_217_728, null);
+    final GraphEdge other = new GraphEdge(-303_054_864, 0, -2_147_483_647);
+
+    // Act
+    final boolean retval = objectUnderTest.equals(other);
+
+    // Assert result
+    Assert.assertEquals(false, retval);
+  }
+
+  @Test
+  public void equalsInputNotNullOutputFalse4() {
+
+    // Arrange
+    final GraphEdge objectUnderTest = new GraphEdge(0, -536_870_913, null);
+    final GraphEdge other = new GraphEdge(null, 1_610_612_735, null);
+
+    // Act
+    final boolean retval = objectUnderTest.equals(other);
+
+    // Assert result
+    Assert.assertEquals(false, retval);
+  }
+
+  @Test
+  public void equalsInputNotNullOutputFalse5() {
+
+    // Arrange
+    final GraphEdge objectUnderTest = new GraphEdge(0, -536_870_913, 0);
+    final GraphEdge other = new GraphEdge(0, -536_870_913, null);
+
+    // Act
+    final boolean retval = objectUnderTest.equals(other);
+
+    // Assert result
+    Assert.assertEquals(false, retval);
+  }
+
+  @Test
+  public void equalsInputNotNullOutputFalse6() {
+
+    // Arrange
+    final GraphEdge objectUnderTest = new GraphEdge(0, -306_783_379, null);
+    final GraphEdge other = new GraphEdge(0, null, null);
+
+    // Act
+    final boolean retval = objectUnderTest.equals(other);
+
+    // Assert result
+    Assert.assertEquals(false, retval);
+  }
+
+  @Test
+  public void equalsInputNotNullOutputFalse7() {
+
+    // Arrange
+    final GraphEdge objectUnderTest = new GraphEdge(0, -536_870_913, null);
+    final GraphEdge other = new GraphEdge(0, 1_610_612_735, null);
+
+    // Act
+    final boolean retval = objectUnderTest.equals(other);
+
+    // Assert result
+    Assert.assertEquals(false, retval);
+  }
+
+  @Test
+  public void equalsInputNullOutputFalse() {
+
+    // Arrange
+    final GraphEdge objectUnderTest = new GraphEdge(null, null, null);
+    final Object other = null;
+
+    // Act
+    final boolean retval = objectUnderTest.equals(other);
+
+    // Assert result
+    Assert.assertEquals(false, retval);
+  }
+
+  @Test
+  public void equalsInputNotNullOutputTrue() {
```

Comment:
Alphabetically, equalsInputNotNullOutputTrue() comes before equalsInputNullOutputFalse() (NotNull before Null) but will anyone notice? I'd suggest not (well, apart from me obviously ;-)

Reply:
A classic case of @petersear :1st_place_medal: